generic_scripts/nlk.py

```diff
@@ -0,0 +1,167 @@
+import numpy as _np
+import matplotlib.pyplot as _plt
+from mathphys.constants import vacuum_permeability
+from mathphys.beam_optics import beam_rigidity
+
+
+class Wire:
+    """."""
+    def __init__(self, position, current):
+        self.pos = position  # Column vector
+        self.curr = current
+
+    def Bfield(self, x, y):
+        mu0 = vacuum_permeability
+        r_w = self.pos  # Wire positions
+        r = _np.array([x, y])[:, None]  # Target position
+        rc = r - r_w  # Cursive r
+        theta = _np.arctan2(rc[1], rc[0])[0]
+        theta_vec = _np.array([-_np.sin(theta), _np.cos(theta)])[:, None]
+        B = mu0 * self.curr/(2*_np.pi*_np.linalg.norm(rc))*theta_vec + r_w
+        return B
+
+    def include_pos_error(self, sigma):
+        self.error = _np.random.normal(loc=0, scale=sigma)
+        self.pos = self.pos + self.error
+
+    @property
+    def pos(self):
+        return self._pos
+
+    @pos.setter
+    def pos(self, position):
+        self._pos = position
+
+    @property
+    def curr(self):
+        return self._curr
+
+    @curr.setter
+    def curr(self, current):
+        self._curr = current
+
+
+class NLK:
+    """."""
+    def __init__(self, curr=1850, errors=False, sigma_errors=1e-5):
+        # Setting wire positions
+        wire_positions = []
```

Comment:
Maybe take `wire_positions` as an input (a list) and make this default list in case of nothing is passed by the user. 
Actually I think it would be nice to set different values of currents for each wire. Maybe this class could accept a dict like `wires_info` containing the lists `wire_positions` and the corresponding `wire_currents`.

Reply:
I think that in the initialization of the object it is not so important to set each current individually. Since we have setters for the current of the wires. Besides, not using a dictionary simplifies the use.